src/js/components/Form/Form.js

```diff
@@ -56,21 +59,29 @@ const Form = forwardRef(
       valueProp,
       valueState,
     ]);
-    const [errors, setErrors] = useState(errorsProp);
-    useEffect(() => setErrors(errorsProp), [errorsProp]);
-    const [infos, setInfos] = useState(infosProp);
-    useEffect(() => setInfos(infosProp), [infosProp]);
     const [touched, setTouched] = useState(defaultTouched);
+    const [validationResults, setValidationResults] = useState(
+      defaultValidationResults,
+    );
+    useEffect(
+      () => setValidationResults({ errors: errorsProp, infos: infosProp }),
+      [errorsProp, infosProp],
+    );
     const validations = useRef({});
 
     // clear any errors when value changes
     useEffect(() => {
-      setErrors(prevErrors => {
+      setValidationResults(prevValidationResults => {
         const [nextErrors] = validate(
```

Comment:
Should this be `[nextErrors, nextInfos]` and then use `nextInfos` at line 83? I would think the `validate()` could return some fresh infos.

Reply:
I think you're right -- we didn't have that prior but I would also think that there could be some new infos coming in after validation